### PR TITLE
CartesianIndexMapper.

### PR DESCRIPTION
**Pull request by dr-robertk**
This PR changes the EclGridManager such that the CartesianIndexMapper is used. 
This way also ALUGrid could be used, e.g. for SPE1 ...

The EclWellManager is changed such that only wells are added that appear to be in the domain at hand.

### CODE REVIEW

File: applications/ebos/eclgridmanager.hh [hunk 1]
```diff
@@ -29,8 +29,11 @@
 #include <ewoms/common/propertysystem.hh>
 #include <ewoms/common/parametersystem.hh>
 
+#include <ewoms/aux/cartesianindexmapper.hh>
 #include <dune/grid/CpGrid.hpp>
 
+//#define EBOS_USE_ALUGRID 1
+
```
Comment (andlaus) [inline applications/ebos/eclgridmanager.hh]: these two lines seem like debugging artifacts

File: applications/ebos/eclwellmanager.hh [hunk 2]
```diff
@@ -108,6 +114,54 @@ public:
...
         }
+
+        for( auto wellIt = wells.begin(); wellIt != wells.end(); ++wellIt )
+        {
+            std::shared_ptr< Well > well = wellIt->second;
+            wellNameToIndex_[ well->name() ] = wells_.size();
+            wells_.push_back( well );
+        }
+
+        return ;
```
Comment (andlaus) [inline applications/ebos/eclwellmanager.hh]: hm, either this `return` statement or the code below seems to be superfluous...

Comment (dr-robertk) [inline applications/ebos/eclwellmanager.hh]: This is for debugging only. The code does not work yet.

**Comment by andlaus** [comment]
thanks for your effors. the PR seems good except for the (probably forgotten) `return` statement. I'll run the usual benchmarks on it tomorrow and if nothing major appears, it will be merged...

**Comment by dr-robertk** [comment]
I have added a bit flexibility to LensGridManager to test the parallelization. With YaspGrid everything works find while with ALUGrid there is an assertion 
globalToDomestic_.find(globalIdx) != globalToDomestic_.end()' failed in globalindices.hh:99

We probably need to discuss this a bit more.

**Comment by andlaus** [comment]
> With YaspGrid everything works find while with ALUGrid there is an assertion
globalToDomestic_.find(globalIdx) != globalToDomestic_.end()' failed in globalindices.hh:99

this looks like some screw-up in the global index code ("global" in the sense that is for all processes, not "global" in the sense of "logically Cartesian"). I'll investigate...

**Comment by dr-robertk** [comment]
You will have to use the paramerge branch from alugrid that contains the parallel 2d version.

**Comment by dr-robertk** [comment]
PS: with only 2 cores it works. Might be a problem in alugrid, who knows.

**Comment by dr-robertk** [comment]
I have removed all changed from the WellManager so that this PR only contains the introduction of the CartesianIndexMapper. From my point of view it's good to be merged.

**Comment by dr-robertk** [comment]
I squashed the commits into 1 commit. It's done now from my point of view.

**Comment by andlaus** [comment]
ok, I'm testing this right now. 1 moment.

**Comment by dr-robertk** [comment]
For got to add cartesianindexmapper.hh, now it should work.

**Comment by andlaus** [comment]
ok, tests finished successfully. merging & thanks.

**Pull request merged**